libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/TestActions/UserEvent.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.TestActions
+{
+    /// <summary>
+    /// Action to script sending event to the bot.
+    /// </summary>
+    [DebuggerDisplay("UserEvent:{Name}")]
+    public class UserEvent : TestAction
```

Comment:
>UserEvent [](start = 17, length = 9)

I suspect a better name would be CustomEvent--it is not actually tied to the user at all.